mesonbuild/interpreter/interpreter.py

```diff
@@ -1706,19 +1714,34 @@ def func_vcs_tag(self, node: mparser.BaseNode, args: T.List['TYPE_var'], kwargs:
                 source_dir,
                 replace_string,
                 regex_selector] + vcs_cmd,
-            'input': kwargs['input'],
-            'output': kwargs['output'],
-            'build_by_default': True,
-            'build_always_stale':True,
-        }
-        return self._func_custom_target_impl(node, [kwargs['output']], cmd_kwargs)
+            self.source_strings_to_files(kwargs['input']),
+            kwargs['output'],
+            build_by_default=True,
+            build_always_stale=True,
+        )
+        self.add_target(tg.name, tg)
+        return tg
 
     @FeatureNew('subdir_done', '0.46.0')
     @noPosargs
     @noKwargs
     def func_subdir_done(self, node, args, kwargs):
         raise SubdirDoneRequest()
 
+    @staticmethod
+    def _validate_custom_target_outputs(node: mparser.BaseNode, has_multi_in: bool,
+                                        outputs: T.Iterable[str], name: str) -> None:
+        """Checks for additional invalid values in a custom_target output.
+
+        This cannot be done with typed_kwargs because it requires the number of
+        inputs.
+        """
+        for out in outputs:
+            if has_multi_in and ('@PLAINNAME@' in out or '@BASENAME@' in out):
+                raise InvalidArguments(f'{name}: output cannot containe "@PLAINNAME@" or "@BASENAME@" '
+                                       'when there is more than one input (we can\'t know which to use)',
+                                       file=node.filename, lineno=node.lineno, colno=node.colno)
```

Comment:
We generally do not pass node information to `InvalidArguments()`, because the interpreter loop already catches exceptions and modifies them to add the current_node information to them.

We do need to pass node information around for Feature checks, but that is because the interpreter doesn't stop everything when it sees a Feature check in order to modify that. :D